src/npc.h

```diff
@@ -118,7 +118,7 @@ class Npc final : public Creature
 		}
 
 		bool isPushable() const final {
-			return walkTicks > 0;
+			return walkTicks > 0 && pushable;
```

Comment:
You're right about the test. I will change the order and use non-equality test then.